scala/README.md

```diff
@@ -525,6 +582,25 @@ It is definitely recommended to read the full Twitter's "[Effective Scala]" guid
     } yield a + b + c
     ```
 
+0. Avoid `isInstanceOf` or `asInstanceOf`. Safe casting is actually one of the best use cases for pattern matching. It is more flexible and allows you, for instance, to use different branches to carry out multiple conditional casts at the same time for various types, perform conversions, or fallback to a default value instead of throwing an exception:
```

Comment:
Maybe point out that pattern matching includes use of `asInstanceOf` and `isInstanceOf` under the hood, but guarantees that a cast will only happen if it can succeed?